lib/kb_ReadsUtilities/kb_ReadsUtilitiesImpl.py

```diff
@@ -81,6 +79,46 @@ def log(self, target, message):
         print(message)
         sys.stdout.flush()
 
+    def add_id_to_plus_line(self,readsfile):
+        gzip_flag = False
+        if readsfile.lower().endswith('.gz'):
+            gzip_flag = True
+
+        if gzip_flag:
+            f = gzip.open(readsfile, 'rt')
+        else:
+            f = open(readsfile, 'r')        
+
+        outputfile = readsfile+".with_IDs"
+        if gzip_flag:
+            outputfile += ".gz"
+            out = gzip.open(outputfile, 'wt')
+        else:
+            out = open(outputfile, 'w')
+    
+        this_id = None
+        counter = 0
+        read_cnt = 0
+        for line in f:
+            line = line.rstrip()
+            if counter == 4:
+                counter = 0
+            if line.startswith('@') and counter == 0:
+                this_id = line.lstrip('@')
+                read_cnt += 1
+                if read_cnt % 1000000 == 0:
+                    print ("reads processed {}".format(read_cnt))
+            if line.startswith('+') and counter == 2 and len(line) < 3:
+                line = '+'+this_id
+            out.write(line+"\n")
+            counter += 1
```

Comment:
is there any library we can utilize for this logic?

Reply:
I appreciate your point about using std libs for basic stuff, but I have about 20 things to get done and adding more things to do when code is already working is low on my priority list.